internal/fs/fs.go

```diff
@@ -2148,6 +2154,13 @@ func (fs *fileSystem) OpenDir(
 	fs.handles[handleID] = handle.NewDirHandle(in, fs.implicitDirs)
 	op.Handle = handleID
 
```

Comment:
I don't think, this is very common or easy to reproduce, considering gcsfuse takes almost 99% time in the first ReadDir() call (7s for 25K files) (since fetches everything from GCS) and next each ReadDir() call takes around 40us to complete. That's why it's hard to give manual interrupt for ReadDir other than the 1st one.

Although I simulated this programmatically and found kernel doesn't cache partial response. Please let me know, this is what you are looking for. Accordingly, I'll update the PR description and internal doc.

Reply:
I meant if ls <dir A> is interrupted after 7th ReadDir call (meaning the 8th and beyond ReadDir calls didn't come). In this case, would the first 7 ReadDir calls output (7 * 2K = 24K) be cached in the kernel's page cache ? If yes, then in next ls <dir A>, are the first 7 ReadDir calls served from page cache and 8th call goes to gcsfuse ? (or there's something else that happens?)

Reply:
Discussed offline:
Conclusion: kernel doesn't cache partial response of a directory.